docs/content/sips/023-build-profiles.md

```diff
@@ -0,0 +1,146 @@
+title = "SIP 023 - Build profiles"
+template = "main"
+date = "2025-03-31T12:00:00Z"
+
+---
+
+Summary: Support defining multiple build profiles for components, and running a Spin application in with those profiles.
+
+Owner(s): [till@fermyon.com](mailto:till@fermyon.com)
+
+Created: March 31, 2025
+
+# Background
+
+Building and running individual components or entire applications in configurations other than the default one is currently difficult to do with Spin: one has to manually edit the `[component.name.build]` tables for some or all components. This is laborious and error-prone, and can lead to accidentally committing and deploying debug/testing/profiling builds to production.
+
+# Proposal
+
+This very simple proposal consists of four parts:
+
+1. Adding an optional `[component.name.profile.profile-name]` table to use for defining additional build profiles of a component
+2. Adding a `-profile [profile-name]` CLI flag and `SPIN_PROFILE` env var to `spin {build, watch, up}` to use specific build profiles of all components (where available, with fallback to the default otherwise)
+   1. By popular demand, the `debug` profile can be selected using the `--debug` CLI flag
```

Comment:
While I added the `--debug` flag for now, I think there might be a reason for not doing so after all: we might want to later add support for actual debugging to Spin, and would in that case probably regret having used this flag for profile selection.

Reply:
@tschneidereit Ah, yep. Reserving `--debug` for "I want to actually attach a debugger" is a good point.  So `spin up --build --debug` would build the debug profile and run under the debugger; but `spin up --build --profile=debug` would build the debug profile but run it without actually debugging.  The first case does feel it should (eventually) be the "common case" and get the fancy flag.